src/transactions/ParallelApplyUtils.cpp

```diff
@@ -630,6 +630,22 @@ GlobalParallelApplyLedgerState::commitChangesToLedgerTxn(
             }
         }
     }
+
+    // We don't need to add the live bucket list restoration keys
```

Comment:
Nit: I think this comment could be clarified a bit: "While the final state of a restored key that will be written to the Live BucketList is already handled in mGlobalEntryMap, we need to let the ltx know what keys need to be removed from the Hot Archive."